WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -130,8 +128,173 @@ final class ProductFormViewController: UIViewController {
+            ServiceLocator.analytics.track(.productDetailShareButtonTapped)
+            self?.displayShareProduct()
+        }
+
+        actionSheet.addDefaultActionWithTitle(ActionSheetStrings.productSettings) { [weak self] _ in
+            ServiceLocator.analytics.track(.productDetailViewSettingsButtonTapped)
+            self?.displayProductSettings()
+        }
+
+        actionSheet.addCancelActionWithTitle(ActionSheetStrings.cancel) { _ in
+        }
+
+        let popoverController = actionSheet.popoverPresentationController
+        popoverController?.barButtonItem = sender
+
+        present(actionSheet, animated: true)
+    }
+
+    // MARK: - UITableViewDelegate
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        let section = tableViewModel.sections[indexPath.section]
+        switch section {
+        case .primaryFields(let rows):
+            let row = rows[indexPath.row]
+            switch row {
+            case .description:
+                ServiceLocator.analytics.track(.productDetailViewProductDescriptionTapped)
+                editProductDescription()
+            default:
+                break
+            }
+        case .settings(let rows):
+            let row = rows[indexPath.row]
+            switch row {
+            case .price:
+                ServiceLocator.analytics.track(.productDetailViewPriceSettingsTapped)
+                editPriceSettings()
+            case .shipping:
+                ServiceLocator.analytics.track(.productDetailViewShippingSettingsTapped)
+                editShippingSettings()
+            case .inventory:
+                ServiceLocator.analytics.track(.productDetailViewInventorySettingsTapped)
+                editInventorySettings()
+            case .categories:
+                // TODO-2509 Edit Product M3 analytics
+                editCategories()
+            case .tags:
+                // TODO-2509 Edit Product M3 analytics
+                editTags()
+            case .briefDescription:
+                ServiceLocator.analytics.track(.productDetailViewShortDescriptionTapped)
+                editBriefDescription()
+            case .externalURL:
+                // TODO-2509 Edit Product M3 analytics
+                editExternalLink()
+                break
+            case .sku:
+                // TODO-2509 Edit Product M3 analytics
+                editSKU()
+                break
+            case .groupedProducts:
+                editGroupedProducts()
+                break
+            case .variations:
+                // TODO-2509 Edit Product M3 analytics
+                guard let product = product as? Product else {
+                    return
+                }
+                guard product.variations.isNotEmpty else {
+                    return
+                }
```

Comment:
Should we merge this into one?
```suggestion
                guard let product = product as? Product, product.variations.isNotEmpty else {
                    return
                }
```

Reply:
yup great suggestion! updated in: https://github.com/woocommerce/woocommerce-ios/pull/2565/commits/55a1f13fa8573d4bf66a0359e920e2acaf78d97f